Firestore/core/test/firebase/firestore/local/index_manager_test.mm

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "Firestore/core/test/firebase/firestore/local/index_manager_test.h"
+
+#include "Firestore/core/src/firebase/firestore/local/index_manager.h"
+#include "Firestore/core/src/firebase/firestore/model/resource_path.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace local {
+
+void IndexManagerTest::AssertParents(std::string collection_id,
+                                     std::vector<std::string> expected) {
+  IndexManager *index_manager = persistence.indexManager;
+  std::vector<model::ResourcePath> actual_paths =
+      index_manager->GetCollectionParents(collection_id);
+  std::vector<std::string> actual;
+  for (model::ResourcePath actual_path : actual_paths) {
+    actual.push_back(actual_path.CanonicalString());
+  }
+  std::sort(expected.begin(), expected.end());
+  std::sort(actual.begin(), actual.end());
+
+  EXPECT_EQ(actual, expected);
```

Comment:
Note that all failures will be reported at this line, regardless of which invocation of `AssertParents` triggered them. One way around it is using [`SCOPED_TRACE`](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#adding-traces-to-assertions).

Reply:
Oh, interesting.  Thanks!